src/Language/PureScript/Docs/Types.hs

```diff
@@ -560,6 +573,7 @@ asDeclaration =
               <*> key "sourceSpan" (perhaps asSourceSpan)
               <*> key "children" (eachInArray asChildDeclaration)
               <*> key "info" asDeclarationInfo
+              <*> key "kind" (perhaps asKindInfo)
```

Comment:
AFAICT, CI is currently failing because of this line. Since the key `kind` doesn't exist in older versions of PS's docs, the parser will fail as it [expects the `kind` key to be there](https://github.com/purescript/purescript/runs/2784351550#step:8:19514):
```
 purescript    >   tests/TestPscPublish.hs:52:17: 
  purescript    >   1) publish, json compatibility, json produced by v0.11.3, symbols-3.0.0.json
  purescript    >        JSON parsing failed: Error in $: At the path: ["modules"][0]["declarations"][0]
  purescript    >        The required key "kind" is missing
  purescript    > 
  purescript    >   To rerun use: --match "/publish/json compatibility/json produced by v0.11.3/symbols-3.0.0.json/"
  purescript    > 
  purescript    >   tests/TestPscPublish.hs:52:17: 
  purescript    >   2) publish, json compatibility, json produced by v0.11.3, generics-4.0.0.json
  purescript    >        JSON parsing failed: Error in $: At the path: ["modules"][0]["declarations"][0]
  purescript    >        The required key "kind" is missing
  purescript    > 
  purescript    >   To rerun use: --match "/publish/json compatibility/json produced by v0.11.3/generics-4.0.0.json/"
  purescript    > 
  purescript    >   tests/TestPscPublish.hs:52:17: 
  purescript    >   3) publish, json compatibility, json produced by v0.12.1, typelevel-prelude-3.0.0.json
  purescript    >        JSON parsing failed: Error in $: At the path: ["modules"][0]["declarations"][0]
  purescript    >        The required key "kind" is missing
  purescript    > 
  purescript    >   To rerun use: --match "/publish/json compatibility/json produced by v0.12.1/typelevel-prelude-3.0.0.json/"
  purescript    > 
```

If I replace `key "kind (perhaps asKindInfo)` with `keyOrDefault "kind" Nothing (Just <$> asKindInfo)`, then I get an error about a type mismatch between an object and null: 
```

purescript>   src/Language/PureScript/Docs/Collect.hs:120:19: 
purescript>   1) publish, preparePackage with json roundtrips, purescript-prelude
purescript>        uncaught exception: ErrorCall
purescript>        An internal error occurred during compilation: Failed to decode: output/Data.Boolean/docs.jsonAt the path: ["declarations"][0]["kind"]
purescript>        Type mismatch:
purescript>        Expected a value of type object
purescript>        Got: null
purescript>        Please report this at https://github.com/purescript/purescript/issues
```
How do I write the parser for `asDeclaration` but make the `kind` key optional?

Reply:
```suggestion
              <*> keyMay "kind" (perhaps asKindInfo)
```

Reply:
Isn't `keyMay "foo" parser` [the same as](https://hackage.haskell.org/package/aeson-better-errors-0.9.1.0/docs/src/Data-Aeson-BetterErrors-Internal.html#keyMay) `keyOrDefault "foo" Nothing (Just <$> parser)`?

Reply:
Oh wait, I see, the problem is that the key could be missing _or_ null _or_ defined. Did you try something like
```suggestion
              <*> keyOrDefault "kind" Nothing (perhaps asKindInfo)
```

Reply:
Thanks! That seems to work.